client.go

```diff
@@ -40,35 +40,69 @@ func init() {
 
 // StatsdClient is a client library to send events to StatsD
 type StatsdClient struct {
-	conn             net.Conn
-	addr             string
-	prefix           string
-	eventStringTpl   string
-	Logger           Logger
-	conn_type        string
-	reconnect_ticker *time.Ticker
+	conn            net.Conn
+	addr            string
+	prefix          string
+	eventStringTpl  string
+	Logger          Logger
+	connType        string
+	reconnectTicker *time.Ticker
 }
 
-// NewStatsdClient - Factory
-func NewStatsdClient(addr string, prefix string) *StatsdClient {
+// ConfigurationFunc is a typedef for a function that configures some aspect of
+// a StatsdClient instance
+type ConfigurationFunc func(*StatsdClient)
+
+// AutoReconnect returns a ConfigurationFunc that causes the StatsdClient to automatically
+// recreate its underlying connection on the specified interval.
+func AutoReconnect(interval time.Duration) ConfigurationFunc {
+	return func(client *StatsdClient) {
+		if interval > 0 {
```

Comment:
is it worth thinking of an exponential backoff vs simple retry every `x` seconds  ?

Reply:
Agreed. This reconnection logic isn't in place to retry something like a failed HTTP (where you might typically see exponential backoff used). It's in place to overcome the need for an application restart on order to get the UDP socket used by the statsd client to get recreated.